src/htm/engine/RegionImpl.cpp

```diff
@@ -105,11 +105,7 @@ void RegionImpl::getParameterArray(const std::string &name, Int64 index, Array &
       NTA_THROW << "setParameterArray: parameter " << name
                 << " does not exist in Spec";
   ParameterSpec p = region_->getSpec()->parameters.getByName(name);
-  if (p.dataType != array.getType()) {
-      NTA_THROW << "setParameterArray: parameter " << name
```

Comment:
is it correct that the setParameterArr is now not used here?

Reply:
> is it correct that the setParameterArr is now not used here?

This is in the parent class of the region implementation.  If the setParameterArray( ) function was not implemented in the subclass it would enter here.   So we can assume that the parameter was not implemented for this region and the throw to complain about the data type is not relevant.